### ISSUE TITLE
World does not generate or load in 1.20

### ISSUE THREAD

**Issue by BrunoGoldbergFerro** [Low-Drag-MC/Shimmer#57]
### Minecraft Version

1.20

### ModLoader

Fabric

### Shimmer version

0.1.15

### Modpack info or mod list(please reduce mod range)

Essential

### The latest.log file and hs_err_pidXXX if exists

no file

### Issue description

The World simply gets stuck at 0% on the loading screen

### Steps to reproduce

Play 1.20 with the mod

**Comment by khjxiaogu** [comment]
Meet same issue
Stack trace:
```
at java.util.concurrent.locks.LockSupport.park(java.base@17.0.3.1/LockSupport.java:211)
        at java.util.concurrent.CompletableFuture$Signaller.block(java.base@17.0.3.1/CompletableFuture.java:1864)
        at java.util.concurrent.ForkJoinPool.compensatedBlock(java.base@17.0.3.1/ForkJoinPool.java:3449)
        at java.util.concurrent.ForkJoinPool.managedBlock(java.base@17.0.3.1/ForkJoinPool.java:3432)
        at java.util.concurrent.CompletableFuture.waitingGet(java.base@17.0.3.1/CompletableFuture.java:1898)
        at java.util.concurrent.CompletableFuture.join(java.base@17.0.3.1/CompletableFuture.java:2117)
        at net.minecraft.server.level.ServerChunkCache.m_7587_(minecraft@1.20.1/ServerChunkCache.java:114)
        at net.minecraft.world.level.Level.m_6522_(minecraft@1.20.1/Level.java:187)
        at net.minecraft.world.level.LevelReader.m_46819_(minecraft@1.20.1/LevelReader.java:129)
        at net.minecraft.world.level.Level.m_6325_(minecraft@1.20.1/Level.java:182)
        at net.minecraft.world.level.Level.m_8055_(minecraft@1.20.1/Level.java:355)
        at com.lowdragmc.shimmer.client.light.LightManager.getLight(shimmer@1.20-0.1.15/LightManager.java:465)
        at net.minecraft.world.level.lighting.BlockLightEngine.handler$zco000$a(minecraft@1.20.1/BlockLightEngine.java:536)
        at net.minecraft.world.level.lighting.BlockLightEngine.m_284436_(minecraft@1.20.1/BlockLightEngine.java)
        at net.minecraft.world.level.lighting.BlockLightEngine.m_75858_(minecraft@1.20.1/BlockLightEngine.java:29)
        at net.minecraft.world.level.lighting.LightEngine.m_9323_(minecraft@1.20.1/LightEngine.java:143)
        at net.minecraft.world.level.lighting.LevelLightEngine.m_9323_(minecraft@1.20.1/LevelLightEngine.java:53)
        at net.minecraft.server.level.ThreadedLevelLightEngine.m_9366_(minecraft@1.20.1/ThreadedLevelLightEngine.java:187)
        at net.minecraft.server.level.ThreadedLevelLightEngine.m_215156_(minecraft@1.20.1/ThreadedLevelLightEngine.java:166)
        at net.minecraft.server.level.ThreadedLevelLightEngine$$Lambda$11811/0x0000000802161fe0.run(minecraft@1.20.1/Unknown Source)
        at net.minecraft.util.thread.ProcessorMailbox.m_18759_(minecraft@1.20.1/ProcessorMailbox.java:91)
        at net.minecraft.util.thread.ProcessorMailbox.m_18747_(minecraft@1.20.1/ProcessorMailbox.java:146)
        at net.minecraft.util.thread.ProcessorMailbox.run(minecraft@1.20.1/ProcessorMailbox.java:102)
        at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(java.base@17.0.3.1/ForkJoinTask.java:1395)
        at java.util.concurrent.ForkJoinTask.doExec(java.base@17.0.3.1/ForkJoinTask.java:373)
        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(java.base@17.0.3.1/ForkJoinPool.java:1182)
        at java.util.concurrent.ForkJoinPool.scan(java.base@17.0.3.1/ForkJoinPool.java:1655)
        at java.util.concurrent.ForkJoinPool.runWorker(java.base@17.0.3.1/ForkJoinPool.java:1622)
        at java.util.concurrent.ForkJoinWorkerThread.run(java.base@17.0.3.1/ForkJoinWorkerThread.java:165)
```
Seems the light engine triggered world generation during world generation, cause a cascading world generation.

**Comment by zomb-676** [comment]
fix since 5ff090f23e0c947ff954c030246d3d0fdeb592dc
